kart/spatial_tree.py

```diff
@@ -214,21 +227,75 @@ def iter_feature_oids(repo, commit_spec):
         )
 
 
-def update_spatial_tree(repo, commit_spec, verbosity=1, clear_existing=False):
+def _minimal_description_of_commit_set(repo, commits):
+    cmd = ["git", "-C", repo.path, "merge-base", "--independent"] + list(commits)
+    try:
+        r = subprocess.run(
+            cmd,
+            encoding="utf8",
+            check=True,
+            capture_output=True,
+            env=tool_environment(),
+        )
+    except subprocess.CalledProcessError as e:
+        raise SubprocessError(
+            f"There was a problem with git show-ref: {e}", called_process_error=e
+        )
+    return set(r.stdout.splitlines())
+
+
+def _build_on_last_index(repo, start_commits, sess):
+    commits_table_exists = sess.scalar(
+        "SELECT count(*) FROM sqlite_master WHERE name = 'commits';"
+    )
+    if commits_table_exists:
+        stop_commits = set(
+            row[0].hex() for row in sess.execute("SELECT commit_id FROM commits;")
+        )
+    else:
+        stop_commits = set()
+
+    all_independent_commits = _minimal_description_of_commit_set(
+        repo, start_commits | stop_commits
+    )
+    start_commits = all_independent_commits - stop_commits
+    return (start_commits, stop_commits, all_independent_commits)
+
+
+def _format_commits(repo, commit_ids):
+    if not commit_ids:
+        return None
+    length = len(repo[next(iter(commit_ids))].short_id)
```

Comment:
```suggestion
    length = len(repo[commit_ids[0]].short_id)
```

Reply:
I'm using lots of sets so they're not subscriptable.